sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/implementation/HttpModelFetcher.java

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.iot.modelsrepository.implementation;
+
+import com.azure.core.exception.AzureException;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.iot.modelsrepository.DtmiConventions;
+import com.azure.iot.modelsrepository.ModelsDependencyResolution;
+import com.azure.iot.modelsrepository.implementation.models.FetchResult;
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * The {@link HttpModelFetcher} is an implementation of {@link ModelFetcher} interface
+ * for supporting http[s] based model content fetching.
+ */
+class HttpModelFetcher implements ModelFetcher {
+    private final ModelsRepositoryAPIImpl protocolLayer;
+    private final ClientLogger logger;
+
+    public HttpModelFetcher(ModelsRepositoryAPIImpl protocolLayer, ClientLogger logger) {
+        this.logger = logger;
+        this.protocolLayer = protocolLayer;
+    }
+
+    @Override
+    public Mono<FetchResult> fetchAsync(String dtmi, URI repositoryUri, ModelsDependencyResolution resolutionOption, Context context) {
+        Queue<String> work = new LinkedList<>();
```

Comment:
`Queue` = `LinkedList`?

Reply:
Queue is an interface, LinkedList is an implementation.